### PR TITLE
[WIP] Fix print toc issue

### PR DESCRIPTION
**Pull request by danduk82**

### CODE REVIEW

**Comment by danduk82** [comment]
not ready yet

**Comment by pfirpfel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- pyramid_oereb/contrib/print_proxy/tocPages.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827&bid=15863702)

File: pyramid_oereb/contrib/print_proxy/tocPages.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            self.compute_d2() + \
+            self.compute_d3() + \
+            self.compute_d4() + \
+            self.compute_d5() + \
+            self.compute_d6()
+        log.warning('total page length : {}'.format(x))
+        return x
+    
+    def getNbPages(self):
+        return -(-self.total_length // self.disposable_height) # ceil number of pages needed
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/print_proxy/tocPages.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827)

File: pyramid_oereb/contrib/print_proxy/tocPages.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    def compute_total_lenght(self):
+        x = self.compute_d1() + \
+            self.compute_d2() + \
+            self.compute_d3() + \
+            self.compute_d4() + \
+            self.compute_d5() + \
+            self.compute_d6()
+        log.warning('total page length : {}'.format(x))
+        return x
+
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/print_proxy/tocPages.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827)

File: pyramid_oereb/contrib/print_proxy/tocPages.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            return self.d3_height
+
+    def compute_d4(self):
+        return self.d4_height
+
+    def compute_d5(self):
+        x = len(self.extract['ThemeWithoutData'] * self.theme_without_data_item_height)
+        if x > self.d5_height:
+            return x
+        else:
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/print_proxy/tocPages.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827)

File: pyramid_oereb/contrib/print_proxy/mapfish_print.py [hunk 4]
```diff
@@ -101,7 +105,8 @@ def __call__(self, value, system):
             'lang': self._language,
             'attributes': extract_as_dict,
         }
-
+        log.debug(json.dumps(extract_as_dict))
+
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/print_proxy/mapfish_print.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827)

File: pyramid_oereb/contrib/print_proxy/tocPages.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            size = self.get_text_size(font_filename, font_size, new_line)
+            if size[0] <= box_width:
+                line.append(word)
+            else:
+                lines.append(line)
+                line = [word]
+        if line:
+            lines.append(line)
+        log.warning('lines : {}'.format(lines))
+        lines = [' '.join(line) for line in lines if line]
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/print_proxy/tocPages.py]: Codacy Issue found: [list comprehension redefines 'line' from line 59](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827)

File: pyramid_oereb/contrib/print_proxy/tocPages.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        self.toc_title_height=15 + 62 + 12 # height + location + item starting position
+        self.toc_item_height=20
+        self.not_concerned_themes_title_height=15 + 26 # height + location
+        self.not_concerned_themes_item_height=12
+        self.theme_without_data_title_height=12
+        self.theme_without_data_item_height=12
+        self.extract = extract
+        self.total_length = self.compute_total_lenght()
+    
+    def get_text_size(self, font_filename, font_size, text):
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/print_proxy/tocPages.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827)

File: pyramid_oereb/contrib/print_proxy/tocPages.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        return self.d4_height
+
+    def compute_d5(self):
+        x = len(self.extract['ThemeWithoutData'] * self.theme_without_data_item_height)
+        if x > self.d5_height:
+            return x
+        else: 
+            return self.d5_height
+
+    def compute_d6_left(self):
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/print_proxy/tocPages.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827)

File: pyramid_oereb/contrib/print_proxy/tocPages.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+from PIL import ImageFont
+
+import textwrap
+import os
+
+log = logging.getLogger(__name__)
+
+class TocPages():
+    """
+    This class is inspired by this example:
```
Comment (pfirpfel) [inline pyramid_oereb/contrib/print_proxy/tocPages.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/pyramid_oereb/pyramid_oereb/pullRequest?prid=4756827)

**Pull request closed without merging**